helpers/wrapped_key.sh

```diff
@@ -20,25 +20,50 @@
 
 set -e
 
-project_id=$1
-location=$2
-terraform_service_account=$3
-keyring=$4
-key=$5
-secret_name=$6
+terraform_service_account=${1}
+key=${2}
+secret_name=${3}
+project_id=${4}
+temporary_crypto_operator_role=${5}
+exe_path=$(dirname ${0})
 
-access_token=$(gcloud auth print-access-token --impersonate-service-account="${terraform_service_account}")
+trap 'catch $? $LINENO' EXIT
+catch() {
+  if    [ "${1}" != "0" ] \
```

Comment:
I think this is checking if a SA string is empty?  If so, it looks odd. I thought there was a differnet way to check.

Should the [[ ]] operator be used, as well?

Reply:
I've double checked this part of the code and did some extra tests. 

The `${1}` parameter which is inside of the trap is being recognized as a system signal (as expected) and not the positional parameter from the variable (`terraform_service_account=${1}`). So, in case of an error in the script, the `${1}` parameter in `[ "${1}" != "0" ]` will receive a non-zero signal. 

If we don't pass the SA we should get an error because the SA is required for the gcloud command. However, the positional parameter from the `terraform_service_account` variable is not related to the`${1}` parameter being used in the catch function. Let me know if you have any questions.

Reply:
Nice!  good to know and thanks for clarifying.